commit 87a0603df5bd99a850321dfb947025bad3f37925
Author: Hannes Mehnert <hannes@mehnert.org>
Date:   Wed Feb 15 22:54:10 2023 +0100

    Allow the private GnuPG key being stored on a SmartCard.
    
    This allows both "sec   " and "sec>  " lines in the output of
    `gpg --batch --list-secret-keys`, the latter being used for private keys stored
    on a SmartCard.
    
    Sponsored-By: Nitrokey GmbH

diff --git a/src/sbs-crypto-signature-gpg.adb b/src/sbs-crypto-signature-gpg.adb
index ca62d9f..0b538cb 100644
--- a/src/sbs-crypto-signature-gpg.adb
+++ b/src/sbs-crypto-signature-gpg.adb
@@ -24,8 +24,10 @@ with GNAT.Regpat;
 package body SBS.Crypto.Signature.GPG
 is
 
+   Sec : constant String := "sec[> ]\s{2}";
+
    Pattern : constant String
-     := "sec\s{3}([a-z]+[1-9][0-9]{3})\s[1-9][0-9]{3}-[0-9]{2}-[0-9]{2}\s\[SC";
+     := Sec & "([a-z]+[1-9][0-9]{3})\s[1-9][0-9]{3}-[0-9]{2}-[0-9]{2}\s\[SC";
 
    -------------------------------------------------------------------------
 
@@ -33,10 +35,13 @@ is
    is
       use type GNAT.Regpat.Match_Location;
 
-      Count : constant Natural
+      Count1 : constant Natural
         := Ada.Strings.Fixed.Count (Source  => Seckey_List,
                                     Pattern => "sec   ");
-
+      Count2 : constant Natural
+        := Ada.Strings.Fixed.Count (Source  => Seckey_List,
+                                    Pattern => "sec>  ");
+      Count : constant Natural := Count1 + Count2;
       Re      : constant GNAT.Regpat.Pattern_Matcher
         := GNAT.Regpat.Compile (Expression => Pattern);
       Matches : GNAT.Regpat.Match_Array (0 .. 1);
